lib/opensusebasetest.pm

```diff
@@ -498,11 +498,18 @@ sub firewall {
 # of wait_boot, e.g. see `git grep -l reboot | xargs grep -L wait_boot`
 sub post_fail_hook {
     my ($self) = @_;
-    return unless $self->{in_wait_boot};
     # In case the system is stuck in shutting down or during boot up, press
     # 'esc' just in case the plymouth splash screen is shown and we can not
     # see any interesting console logs.
-    send_key 'esc';
+    if ($self->{in_wait_boot}) {
+        send_key 'esc';
+        save_screenshot;
+    }
+    # Mount shared memory to /tmp if detected system is Read-Only
+    else {
```

Comment:
I think an early-return from the above if-branch would be cleaner to read than the else-branch, WDYT?